### ISSUE TITLE
[Model]Runtime error happens during model compilation

### ISSUE THREAD

**Issue by yszhou2019** [openvinotoolkit/openvino#14028]
##### System information (version)
- OpenVINO => 2022.2.0
- Operating System / Platform => almalinux8
- Problem classification: Model Conversion
- Framework: pytorch
- Language: python

##### Detailed description
I've already converted the torchscript model to onnx format and successfully run it with onnxruntime. I try to use openvino for comparison and runtime error was thrown during model compilation.

I have no idea what's happening. Can anyone help with this?
```
Traceback (most recent call last):
  File "/app/tests/ov.py", line 78, in <module>
    ov_model = OpenVINODynamicWrapper(encoder_path, decoder_path)
  File "/app/src/utils/ov_utils.py", line 37, in __init__
    self.decoder = core.compile_model(decoder_model, "CPU")
  File "/usr/local/lib64/python3.9/site-packages/openvino/runtime/ie_api.py", line 387, in compile_model
    super().compile_model(model, device_name, {} if config is None else config),
RuntimeError: Check 'args_et.is_dynamic() || args_et != element::boolean' failed at core/src/op/util/binary_elementwise_arithmetic.cpp:36:
While validating node 'v1::Multiply Multiply_46166 (Unsqueeze_47[0]:boolean{?,1,?}, Constant_46165[0]:boolean{1,1,1}) -> (dynamic...)' with friendly_name 'Multiply_46166':
Arguments cannot have boolean element type (argument element type: boolean).
```

##### Steps to reproduce
```python
import time
import torch
import openvino.runtime as ov

class OpenVINODynamicWrapper:
    
    def __init__(self, encoder_path: str, decoder_path: str, device: torch.device = torch.device("cpu")) -> None:
        self.device = device
        self.encoder_path = encoder_path
        self.decoder_path = decoder_path

        core = ov.Core()

        encoder_model = core.read_model(encoder_path)
        encoder_shapes = [
            [ov.Dimension(), ov.Dimension(), 40],
            [ov.Dimension(), ov.Dimension()],
        ]
        partial_shapes = { index: ov.PartialShape(shape) for index, shape in enumerate(encoder_shapes) }
        encoder_model.reshape(partial_shapes)
        self.encoder = core.compile_model(encoder_model, "CPU")

        decoder_model = core.read_model(decoder_path)
        decoder_shapes = [
            [ov.Dimension(), ov.Dimension(), 1024],
            [ov.Dimension(), ov.Dimension()],
            [ov.Dimension(), 1],
            [2, ov.Dimension(), 1024],
            [2, ov.Dimension(), 1024],
            [ov.Dimension(), ov.Dimension()],
            [ov.Dimension()],
            [1, ov.Dimension(), 1024],
            [1, ov.Dimension(), 1024],
        ]
        partial_shapes = { index: ov.PartialShape(shape) for index, shape in enumerate(decoder_shapes) }
        decoder_model.reshape(partial_shapes)
        self.decoder = core.compile_model(decoder_model, "CPU")

    def encode(self, src_seq, src_mask, hid=None):
        infer_req = self.encoder.create_infer_request()
        infer_req.set_input_tensor(0, src_seq)
        infer_req.set_input_tensor(1, src_mask)
        s=time.time()
        infer_req.infer()
        print(f"infer {time.time()-s} s")
        return infer_req.get_output_tensor(0), infer_req.get_output_tensor(1)

    def decode(self, enc_out, enc_mask, tgt_seq, hid=None, cell=None, padded_bias_phases=None,
               bias_phases_batch=None, bias_hid=None, bias_cell=None):
        infer_req = self.decoder.create_infer_request()
        infer_req.set_input_tensor(0, enc_out)
        infer_req.set_input_tensor(1, enc_mask)
        infer_req.set_input_tensor(2, tgt_seq)
        infer_req.set_input_tensor(3, hid)
        infer_req.set_input_tensor(4, cell)
        infer_req.set_input_tensor(5, padded_bias_phases)
        infer_req.set_input_tensor(6, bias_phases_batch)
        infer_req.set_input_tensor(7, bias_hid)
        infer_req.set_input_tensor(8, bias_cell)
        infer_req.infer()
        return infer_req.get_output_tensor(0), infer_req.get_output_tensor(1), infer_req.get_output_tensor(2), infer_req.get_output_tensor(3)

encoder_path = "encoder.onnx"
decoder_path = "decoder.onnx"
ov_model = OpenVINODynamicWrapper(encoder_path, decoder_path)

NUMPY_TYPE_TO_OV_TYPE = {
    np.uint8: ov.Type.u8,
    np.int32: ov.Type.i32,
    np.int64: ov.Type.i64,
    np.float32: ov.Type.f32,
}

def construct_np_encode_input():
    src_seq = np.random.randn(1, 60054, 40).astype(np.float32)
    src_mask = np.random.randn(1, 60054).astype(np.uint8)
    return src_seq, src_mask

def construct_np_decode_input():
    enc_out = np.random.randn(1, 15012, 1024).astype(np.float32)
    enc_mask = np.random.randn(1, 15012).astype(np.uint8)
    tgt_seq = np.random.randn(1, 1).astype(np.int64)
    hid = np.random.randn(2, 1, 1024).astype(np.float32)
    cell = np.random.randn(2, 1, 1024).astype(np.float32)
    padded_bias_phases = np.random.randn(3, 1).astype(np.int32)
    bias_phases_batch = np.random.randn(3).astype(np.float32)
    bias_hid = np.random.randn(1, 3, 1024).astype(np.float32)
    bias_cell = np.random.randn(1, 3, 1024).astype(np.float32)
    return enc_out, enc_mask, tgt_seq, hid, cell, padded_bias_phases, bias_phases_batch, bias_hid, bias_cell

def to_ov_tensor(inputs):
    return tuple([ov.Tensor(input, list(input.shape), NUMPY_TYPE_TO_OV_TYPE[input.dtype.type]) for input in inputs])

def ov_encode():
    input = to_ov_tensor(construct_np_encode_input())
    start = time.time()
    for i in range(10):
        begin = time.time()
        res = ov_model.encode(*input)
        print(f"{i+1} iteration takes {time.time() - begin} s")
    print(f"ov encode avg : {(time.time() - start)/10} s")
    print(res)

def ov_decode():
    input = to_ov_tensor(construct_np_decode_input())
    start = time.time()
    for i in range(10):
        begin = time.time()
        res = ov_model.decode(*input)
        print(f"{i+1} iteration takes {time.time() - begin} s")
    print(f"ov decode avg : {(time.time() - start)/10} s")
    print(res)

ov_encode()
ov_decode()
```

I can send you the randomized model, please leave your email.

![image](https://user-images.githubusercontent.com/59953951/202132542-d5a88d2e-a6c2-4480-80c3-293739450590.png)

**Comment by rkazants** [comment]
Hi @mlukasze,

I think we should resolve this problem on the ONNX FE side. Currently, Mul operation from ONNX specification should support input boolean type (despite it is not mentioned in the ONNX specification - it successfully inferred via onnx-runtime). My proposal to extend `Mul` translator in ONNX FE and in case of input boolean type (just check it by `get_element_type()` method) we need to translate it to `LogicalAnd` (from our opset) instead of `Multiply`.

Best regards,
Roman

**Comment by mitruska** [comment]
Hello, I've created a PR #14047 to enable conversion of ONNX Mul with boolean inputs to OV LogicalAnd according to @rkazants suggestion. 

- https://github.com/openvinotoolkit/openvino/pull/14047

But what I can see in the attached to the issue logs is that the `read_model` step was successful and the error comes from the `compile_model` after `decoder_model.reshape`.

@yszhou2019  If you would like to share the randomized model for further validation on our side, you can find our emails on our github profiles or you can share a link to the model in the comment.

**Comment by andrei-kochin** [comment]
@mitruska looks like @eaidova faced similar issue during model research so she can share model with you as well

**Comment by yszhou2019** [comment]
@mitruska All right. I've already sent the model to you. Thank you!

**Comment by mitruska** [comment]
I was able to reproduce the issue at the `compile_model` step. 
There is no boolean multiplication in the original onnx model, this is why `read_model` step was succesful. 
It means the change in OV ONNX FE (PR [#14047](https://github.com/openvinotoolkit/openvino/pull/14047)) is not needed.

The error comes from the Broadcast-3 transformation, so I've created another PR with the corresponding fix:
- https://github.com/openvinotoolkit/openvino/pull/14085

@yszhou2019 Thank you for reporting this issue!

**Comment by yszhou2019** [comment]
Hi @mitruska Do you have any idea how I can apply openvino 2022.2.0 with this dynamic model? Or, will this problem be fixed in next release?

**Comment by mitruska** [comment]
@yszhou2019 The PR https://github.com/openvinotoolkit/openvino/pull/14085 is going to be merged soon, but the fix is not a part of pre-release package yet. 
We will keep you informed:)

**Comment by mitruska** [comment]
The PR https://github.com/openvinotoolkit/openvino/pull/14085 resolving this issue has been merged to the openvino master branch, 
and the fix will be included in OpenVINO `2022.3` release.
@yszhou2019 Let us know if there's anything more we can do to help!
